File: app/src/types/request/report.ts
// Copyright (C) Call of Nil contributors
// SPDX-License-Identifier: AGPL-3.0-only

import type { WorldId } from '@/types/core/world';
import type { PlayerId } from '@/types/core/player';
import type { ReportId } from '@/types/core/report';

export interface ForwardReportRequest {
  readonly world: WorldId;
  readonly id: ReportId;
  readonly recipient: PlayerId;
}

export interface GetReportRequest {
  readonly world: WorldId;
  readonly id: ReportId;
}

export interface GetReportsRequest {
  readonly world: WorldId;
  readonly ids: readonly ReportId[];
  readonly limit: Option<number>;
}

File: crates/nil-payload/src/request/report.rs
// Copyright (C) Call of Nil contributors
// SPDX-License-Identifier: AGPL-3.0-only

use bon::Builder;
use nil_core::player::PlayerId;
use nil_core::report::ReportKind;
use nil_core::world::config::WorldId;
use serde::{Deserialize, Serialize};

#[derive(Builder, Clone, Debug, Deserialize, Serialize)]
#[serde(rename_all = "camelCase")]
#[cfg_attr(feature = "typescript", derive(ts_rs::TS))]
#[cfg_attr(feature = "typescript", ts(export))]
pub struct ForwardReportRequest {
  #[builder(start_fn, into)]
  pub world: WorldId,
  #[builder(into)]
  pub recipient: PlayerId,
  #[builder(into)]
  pub report: ReportKind,
}
